src/ui/RelevanceInspector/RelevanceInspector.ts

```diff
@@ -0,0 +1,171 @@
+      (e, args: IAttributeChangedEventArg) => this.toggleFromState(args.value)
+    );
+    $$(this.element).on('click', () => this.open());
+    $$(this.bindings.root).on(ResultListEvents.newResultDisplayed, (e, args: IDisplayedNewResultEventArgs) =>
+      this.handleNewResultDisplayed(args)
+    );
+    this.bindings.queryStateModel.get(QueryStateModel.attributesEnum.debug) ? this.show() : this.hide();
+  }
+
+  public get modalBox() {
+    return this.modalBoxModule;
+  }
+
+  public set modalBox(modalBoxModule) {
+    this.modalBoxModule = modalBoxModule;
+  }
+
+  public hide() {
+    $$(this.element).addClass('coveo-hidden');
+  }
+
+  public show() {
+    $$(this.element).removeClass('coveo-hidden');
+  }
+
+  public async open() {
+    if (this.opened) {
+      return;
+    }
+    this.opened = true;
+
+    const content = $$('div');
+    const animation = DomUtils.getBasicLoadingAnimation();
+    content.append(animation);
+    this.modalBox.open(content.el, this.modalBoxOptions);
+    const rows = await this.buildTabs();
+    if (rows) {
+      animation.remove();
+      content.append(rows.el);
+    } else {
+      this.modalBox.close();
+    }
+  }
+
+  private handleNewResultDisplayed(args: IDisplayedNewResultEventArgs) {
+    if (this.bindings.queryStateModel.get(QueryStateModel.attributesEnum.debug)) {
+      $$(args.item).addClass('coveo-with-inline-ranking-info');
+      if ($$(args.item).hasClass('coveo-table-layout')) {
+        $$(args.item).append(new InlineRankingInfo(args.result).build().el);
+      } else {
+        $$(args.item).prepend(new InlineRankingInfo(args.result).build().el);
+      }
+    }
+  }
+
+  private get modalBoxOptions() {
+    return {
+      title: 'Relevance Inspector',
+      titleClose: false,
+      overlayClose: true,
+      sizeMod: 'big',
+      className: 'relevance-inspector-modal',
+      validation: () => {
+        this.opened = false;
+        return true;
+      }
+    };
+  }
+
+  private toggleFromState(stateValue: number) {
+    stateValue != 0 ? this.show() : this.hide();
+    this.modalBox.close();
+  }
+
+  private async buildTabs() {
+    const lastResults = this.bindings.queryController.getLastResults().results;
+    if (!lastResults || lastResults.length == 0) {
+      new Logger(this).error(
+        'Could not open Relevance Inspector because there is no results to display. Please execute a query beforehand',
+        lastResults
+      );
+      return;
+    }
+    if (!lastResults[0].rankingInfo) {
+      new Logger(this).error(
+        'Could not open Relevance Inspector because there is no ranking info returned on the results. Please execute a query in debug mode beforehad',
+        lastResults
+      );
+      return;
+    }
+
+    const container = $$('div');
+
+    const rankingInfoTable = await new RankingInfoTable(lastResults, this.bindings);
```

Comment:
Do we need the await keyword here?

Reply:
No dont need them,I forgot to remove them when I refactored that part. 

Kinda weird that the TS compiler allows you to use the await keyword on a function that does not return a promise ... 🤔

Reply:
It's in the spec of the await keyword. Not sure why but using await on a non promise value will return a promise that resolves immediately.